### PR TITLE
enhance: Use fdopen, fwrite to reduce direct syscall

### PR DESCRIPTION
**Pull request by congqixia**
`File.Write` and `File.WriteInt` use `write`, which may be just direct syscall in some systems. When mappding field data and write line by line, this could cost lost of CPU time when the row number is large.

### CODE REVIEW

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
Here is some benchmark result for FWrite APIs vs origin Write ones
----------------------------------------------------------------
Benchmark                      Time             CPU   Iterations
----------------------------------------------------------------
BN_File_Write_Syscall       3923 ns         3922 ns       139648
BN_FILE_Write_Stream        46.6 ns         46.6 ns     14594626

File: internal/core/src/common/File.h [hunk 1]
```diff
@@ -27,8 +29,8 @@ class File {
         file.fd_ = -1;
     }
     ~File() {
-        if (fd_ >= 0) {
-            close(fd_);
+        if (fs_ != nullptr) {
+            fclose(fs_);
```
Comment (zhagnlu) [inline internal/core/src/common/File.h:33]: fclose(fs_); -> Close() is better

File: internal/core/src/common/File.h [hunk 2]
```diff
@@ -63,22 +65,45 @@ class File {
         return write(fd_, &value, sizeof(value));
```
Comment (liliu-z) [inline internal/core/src/common/File.h:65]: Why don't replace this directly

Comment (congqixia) [reply internal/core/src/common/File.h:65]: need add fflush for every write usage

**Pull request merged**